### ISSUE TITLE
"Cool - cleaning" and "Dry - cleaning" should show as off

### ISSUE THREAD

**Issue by Sebazzz** [h4de5/home-assistant-toshiba_ac#145]
My suggestion is to rework both "cleaning" modes of "dry" and "cool".

**Background**
For those who don't know, "cleaning" is a state the aircon unit is in to vapourize any leftover moisture after either drying or cooling to prevent mold growth. Then you turn the unit off after having it in either cool or dry it switches off, then on again in the state of "[cool|dry] - cleaning".

**The issue**
However, as far as Home Assistant and any automation or integration is concerned the aircon is still on. That can automations to misfire because rarely you're interested whether it is cleaning mode: you just care whether it is cooling or is off. Also, in for instance an limited integration like HomeKit or certain HA widgets, you can't even see that it is in cleaning mode because it doesn't show it. The user will then turn the aircon off again, thinking it didn't respond (as the wifi units themselves are already quite unstable sometimes), and the cleaning mode is terminated leading to mold growth.

**Suggestion**
When the unit goes to cleaning mode, actually set the status to "off". Add an optional diagnostic binary_sensor that can show cleaning to be either "on" or "off" when cleaning is engaged. This has the result that things can still be automated based on the cleaning mode if so desired, but as far as Home Assistant and any integration is concerned the unit is already off - even if it is still cleaning.

When the user then desires to have the unit in a different mode, the integration should then send an "off" command to terminate the cleaning and then the command the user has chosen (otherwise the unit will not respond until cleaning has terminated).

**Comment by h4de5** [comment]
I think if HA thinks the climate is off already, it will not send an off command again. so the last part won't work. 
besides that, I would need to have a look if cleaning modes are coming back from the api as separate state.

**Comment by Sebazzz** [comment]
> I think if HA thinks the climate is off already, it will not send an off command again. so the last part won't work.
besides that, I would need to have a look if cleaning modes are coming back from the api as separate state.

String matching might work. but at the integration side it is in control itself right. The integration itself can of course directly talk to the Toshiba API.

**Comment by Sebazzz** [comment]
> @Sebazzz: have you tried adjusting your automations or created your own template sensors to reflect the state you desire using the actual state and attributes as currently returned by the integration?
> 
> I think leaving the integration to reflect the actual state would be a cleaner approach and allow more flexibility for other users to create their own automations per their individual needs.

No, I haven't tried it because I wasn't aware of the attributes.

Still, when it is in cleaning, the actual state should be off then. It is not producing cold air or contributing to drying the air in the room. It makes much more sense from an user point of view. That it is cleaning is just an detail that prevents mold, but it is not actually on.

**Comment by Sebazzz** [comment]
Yes, but it is not just a matter of taste :) Turning the unit off (again, because your integration turned it off - now you turn it off thinking it is still on) will terminate cleaning mode and could mold growth and a smell.

**Comment by markkuleinio** [comment]
I'd really like this integration to show the real cleaning mode of the device.

The Toshiba app shows the device in cleaning mode:

<img width="655" height="772" alt="Image" src="https://github.com/user-attachments/assets/c0bef011-d4ee-484c-8d9b-78b6970d98c7" />

The same device in HA thermostat is still shown in cooling mode (incorrectly suggesting that the user should turn it off):

<img width="673" height="859" alt="Image" src="https://github.com/user-attachments/assets/bf223073-10c5-40ca-9333-6e493648f262" />

**Comment by h4de5** [comment]
https://developers.home-assistant.io/docs/core/entity/climate/#hvac-action

at this point we cannot define our own states. "idle" would come closest to what you need. but this could led to confusion when target temperature is simply reached.

**Comment by markkuleinio** [comment]
I see. How about showing the cleaning mode as "fan"? (It is a fan-only mode actually, just with reduced power and direction) I mean, the currently shown "cool" is incorrect anyway, and if the user has used the device in cooling and tried to stop it (and the device goes to cleaning mode), then "fan" at least signals that the stop request has already reached the device.

**Comment by markkuleinio** [comment]
Any further comments or ideas? I'm reluctant to build additional logic to first detect the "off" command (issued by automation or a family member) and then prevent any additional "off" commands (again either by automation or a family member) during the next 30 minutes. If the cleaning mode would be presented with anything else than the current "on", then it would be trivial to just ignore "off" command by anyone if the AC is not currently "on".
